oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DetailedRevisionGCStatsCollector.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.plugins.document;
+
+import org.apache.jackrabbit.oak.plugins.document.VersionGarbageCollector.VersionGCStats;
+
+/**
+ * Collector interface for {@link DocumentNodeStore} Detailed revision garbage collection
+ * statistics.
+ */
+public interface DetailedRevisionGCStatsCollector {
+
+    void documentRead();
+
+    void deletedPropertiesRead(long numProps);
```

Comment:
Do we need this level of granularity?

I understand that `documentsUpdated` might not contain the full set of documents we wanted to update, as `findAndUpdate` might fail for some. 

But I'm wondering if we rather should have a counter for those failures on the document granularity level, rather than deeper (eg deletedPropertiesRead, but also the same for unmerged branch commits and individual revisions). Wdyt?

Reply:
I agree that we should have a counter for documents which had not been updated. I will update the PR.

> Do we need this level of granularity?

Collecting low-level metrics might help us understand the nature of the garbage that we are missing along with the number of documents.

However, In case we find this too noisy, we can stop collecting them.

Reply:
We do have `propertiesDeleted` though, which is a low-level metric, ie does provide the granularity of how many properties were deleted. The `deletedPropertiesRead` I'm referring to here is usually identical to `propertiesDeleted` unless `findAndUpdate` fails (which should not fail frequently given we mostly update old document).

I'm just asking whether we need it, as I suspect it is nearly a duplicate of `propertiesDeleted`?